src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -244,12 +247,18 @@ class ReportActionItemMessageEdit extends React.Component {
                             toggleReportActionComposeView(false, this.props.isSmallScreenWidth);
                         }}
                         onBlur={(event) => {
+                            const relatedTargetId = lodashGet(event, 'nativeEvent.relatedTarget.id');
+
                             // Return to prevent re-render when save/cancel button is pressed which cancels the onPress event by re-rendering
-                            if (_.contains([this.saveButtonID, this.cancelButtonID, this.emojiButtonID], lodashGet(event, 'nativeEvent.relatedTarget.id'))) {
+                            if (_.contains([this.saveButtonID, this.cancelButtonID, this.emojiButtonID], relatedTargetId)) {
                                 return;
                             }
                             this.setState({isFocused: false});
-                            toggleReportActionComposeView(true, this.props.isSmallScreenWidth);
+
+                            if (this.messageEditInput === relatedTargetId) {
+                                return;
+                            }
+                            openReportActionComposeViewWhenClosingMessageEdit(this.props.isSmallScreenWidth);
```

Comment:
Why do we still need to delay opening the main composer? Isn't the `relatedTargetId` check enough?

Reply:
@mananjadhav @tienifr